posthog/api/event.py

```diff
@@ -212,9 +212,11 @@ def actions(self, request: request.Request) -> response.Response:
             if event.pk in ids_seen:
                 continue
             ids_seen.append(event.pk)
-            for action in event.action_set.filter(deleted=False):
-                event.action = action
-                matches.append(event)
+            for action in event.action_set.all():
+                # do not use .filter(), because it causes N+1 query problem. Filter just in memory
```

Comment:
Thanks for this! An easier solution might be to do this filtering in Prefetch on line 207:
` .prefetch_related(Prefetch("action_set", queryset=Action.objects.filter(deleted=False).order_by("id")))[0:101]`

I think that should work (thought could be wrong!)

Reply:
Thanks for the tip, tried it, looks like it works correctly